js/ui/handlers.js

```diff
@@ -149,5 +161,43 @@ function Handlers(map) {
                 map.rotating = false;
                 map._rerender();
             }, 200);
-        });
+        })
+        .on('boxzoomstart', function(e) {
+            if (!box) {
+                box = DOM.create('div', 'mapboxgl-boxzoom', map.getContainer());
+                map.getContainer().classList.add('mapboxgl-crosshair');
+                map.fire('boxzoomstart');
+            }
+
+            function getBounds(a, b) {
+                return {
+                    max: {
+                      x: Math.max(a.x, b.x),
+                      y: Math.max(a.y, b.y)
+                    },
+                    min: {
+                      x: Math.min(a.x, b.x),
+                      y: Math.min(a.y, b.y)
+                    }
+                };
+            }
+
+            var bounds = getBounds(e.current, e.start);
+            DOM.setTransform(box, 'translate(' + bounds.min.x + 'px,' + bounds.min.y + 'px)');
+            box.style.width = (bounds.max.x - bounds.min.x) + 'px';
+            box.style.height = (bounds.max.y - bounds.min.y) + 'px';
+        })
+        .on('boxzoomend', function(e) {
+            boxzoomFinish();
+
+            var bounds = new LatLngBounds(
+                map.containerPointToLatLng(e.start),
+                map.containerPointToLatLng(e.current)
+            );
+
+            map.fitBounds(bounds).fire('boxzoomend', {
```

Comment:
you could also pass `linear: true` in `fitBounds` options for a more intuitive easing in this case.